src/reseeding.rs

```diff
@@ -83,63 +83,43 @@ impl<R: RngCore + SeedableRng, Rsdr: RngCore> ReseedingRng<R, Rsdr> {
     /// Reseed the internal PRNG.
     ///
     /// This will try to work around errors in the RNG used for reseeding
-    /// intelligently. If the error kind indicates retrying might help, it will
-    /// immediately retry a couple of times. If the error kind indicates the
-    /// seeding RNG is not ready, it will retry later, after `threshold / 256`
-    /// generated bytes. On other errors in the source RNG, this will skip
-    /// reseeding and continue using the internal PRNG, until another
-    /// `threshold` bytes have been generated (at which point it will try
-    /// reseeding again).
-    #[inline(never)]
+    /// intelligently through some combination of retrying and delaying
+    /// reseeding until later. So long as the internal PRNG doesn't fail, this
+    /// method will not fail, i.e. failures from the reseeding source are not
+    /// fatal.
     pub fn reseed(&mut self) {
-        trace!("Reseeding RNG after generating {} bytes",
-               self.threshold - self.bytes_until_reseed);
-        self.bytes_until_reseed = self.threshold;
-        let mut err_count = 0;
-        loop {
-            if let Err(e) = R::from_rng(&mut self.reseeder)
-                            .map(|result| self.rng = result) {
-                let kind = e.kind();
-                if kind.should_wait() {
-                    self.bytes_until_reseed = self.threshold >> 8;
-                    warn!("Reseeding RNG delayed for {} bytes",
-                           self.bytes_until_reseed);
-                } else if kind.should_retry() {
-                    err_count += 1;
-                    // Retry immediately for 5 times (arbitrary limit)
-                    if err_count <= 5 { continue; }
-                }
-                warn!("Reseeding RNG failed; continuing without reseeding. Error: {}", e);
-            }
-            break; // Successfully reseeded, delayed, or given up.
-        }
+        // Behaviour is identical to `try_reseed`; we just squelch the error.
+        let _res = self.try_reseed();
```

Comment:
Nice clean-up, and I think I can get behind the choice to unify the two reseeding methods. But I think doing the retries immediately is something useful to keep. At least if the error is `Transient`, with `RDRAND` in mind.

Reply:
You think? I don't expect `RDRAND` would report an error *often*, and we're only delaying until the next call, so I thought it was reasonable?

Reply:
Doesn't it wait until `self.threshold >> 8`, which could be large? But okay, I think this is a reasonable solution. The time to pick between reseeds is not exact science anyway...

Reply:
No, the code sets `bytes_until_reseed = 0` in case of `Transient` errors.